### DISCUSSION TITLE
Solving the laplace equation in only one direction while in a 2 or 3D domain

### DISCUSSION THREAD

**Discussion by Tobias-Dwyer** [zwicker-group/py-pde#199] [Q&A] [answered]
Hello I was wondering if it was possible to solve the Laplace in only one direction while in a 3D (x,y, z) domain. 

For my current problem this means that I have a generating function that works over the entire 3D domain but diffusion only occurs in the z direction resulting in an equation that looks sort of like this. 
![image](https://user-images.githubusercontent.com/55504950/150052816-1437b51c-3982-46d6-8f7c-065eea23ce50.png)

The current implementation solves for diffusion in all spatial directions but I would love a workaround if it's possible. Thanks! :D

**Comment by david-zwicker** [comment, marked as answer]
Yes, this is possible, although it is currently not very well exposed in the general framework. Here is an example that should get you started:

```python
from pde.grids.operators.cartesian import _make_derivative, _make_derivative2

make_derivative_x = functools.partial(_make_derivative, axis=0)
pde.CartesianGrid.register_operator('derivative_x', make_derivative_x)
make_laplace_x = functools.partial(_make_derivative2, axis=0)
pde.CartesianGrid.register_operator('laplace_x', make_laplace_x)

eq = pde.PDE({'u': 'derivative_x(g/D*derivative_x(u))'})
```

The example defines two new operators, `derivative_x` and `laplace_x`, that can be used in the expressions of the `PDE` class. They evaluate the first and second derivative along the `x`-direction, respectively.

**Reply by Tobias-Dwyer** [reply]
This is sweet, thanks for the quick response!
